data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java

```diff
@@ -56,27 +61,33 @@ public ScanBuilder reuseContainers() {
     }
 
     public ScanBuilder where(Expression rowFilter) {
-      this.where = Expressions.and(where, rowFilter);
+      this.tableScan = this.tableScan.filter(Expressions.and(defaultWhere, rowFilter));
```

Comment:
The `TableScan` API is a refinement API, like Spark's data frames. When you filter, you get a new scan that has the new filter applied with the existing scan's filters. So there's no need to use `Expressions.and`. In fact, using `and` with `true` and another filter will just return the filter.

Reply:
Seems like we can either keeping refining the `tableScan` obj in each of the builder methods in which case we might not need to have instance fields like `where`, `caseSensitive` etc and we should delete them where possible. Or we create the `tablescan` once in the `build` method in which case we will need to keep our own instance fields, but then we should remove the tablescan refine from each fo the build methods. Either approach is fine by me.

Reply:
Agreed. I just don't think we should have two sets of scan defaults.